piazza-analytics-chrome/src/inject.js

```diff
@@ -1,21 +1,38 @@
-var readyStateCheckInterval = setInterval(function() {
-    if (document.readyState === "complete") {
-        clearInterval(readyStateCheckInterval);
-        $(window).hashchange(updateNames);
+var lastPiazzaPost = -1;
 
-        updateNames()
+$(document).ready(function() {
+    $(window).unload(updateNames);
+    // wait until a dom thing exists
+    updateNames();
+});
 
-    }
-}, 50);
+new MutationObserver(function(mutations) {
+    mutations.forEach(function(mutation) {
+        if (!mutation.addedNodes) {
+            return;
+        }
+        updateNames();
+    });
+}).observe(document.body, {
+    childList: true,
+    subtree: true,
+    attributes: false,
+    characterData: false
+});
 
 function updateNames() {
-    var server_ip = "https://cdn.moe/piazza/"
+    var proxyUrl = "https://cdn.moe/piazza/";
```

Comment:
oops my bad. I haven't done js in a while if it's not obvious... python for dayzz